src/tools/inner_node.rs

```diff
@@ -48,7 +52,11 @@ impl InnerNode {
         // generate the keypair and prepare the crypto engine
 
         let engine = Secp256k1::new();
-        let (private_key, public_key) = engine.generate_keypair(&mut secp256k1::rand::thread_rng());
+        let (private_key, public_key) = if config.synth_node_config.generate_new_keys {
+            engine.generate_keypair(&mut secp256k1::rand::thread_rng())
+        } else {
+            decode_keys().expect("Invalid predefined keys.")
```

Comment:
```suggestion
            decode_keys().expect("invalid predefined keys")
```
Check:
https://github.com/runziggurat/algorand/pull/14#discussion_r1001654710

Reply:
Done